commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,160 @@
+        ignoreFocusOut: false,
+        placeHolder: 'Basic',
+        value: 'Basic',
+        prompt: 'SKU? '
+    });
+
+    client.registries.beginCreate(resourceGroup.name, registryName, { 'sku': { 'name': sku }, 'location': resourceGroup.location }).then(function (response) {
+        vscode.window.showInformationMessage(response.name + ' has been created succesfully!');
+    }, function (error) {
+        vscode.window.showErrorMessage(error.message);
+    })
+
+    //Acquiring telemetry data here
+    if (reporter) {
+        /* __GDPR__
+           "command" : {
+              "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+           }
+         */
+        reporter.sendTelemetryEvent('command', {
+            command: teleCmdId
+        });
+
+        if (registryName.toLowerCase().indexOf('azurecr.io')) {
+            /* __GDPR__
+               "command" : {
+                  "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+               }
+             */
+            reporter.sendTelemetryEvent('command', {
+                command: teleAzureId
+            });
+        }
+    }
+
+}
+
+// INPUT HELPERS
+async function acquireSubscription(): Promise<SubscriptionModels.Subscription> {
+    let subscription: SubscriptionModels.Subscription;
+    const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+
+    let subsNames: string[] = [];
+    for (let i = 0; i < subs.length; i++) {
+        subsNames.push(subs[i].displayName);
+    }
+    let subscriptionName: string;
+    do {
+        subscriptionName = await vscode.window.showQuickPick(subsNames, { 'canPickMany': false, 'placeHolder': 'Choose a subscription to be used' });
+
+        if (subscriptionName === undefined) throw 'User exit';
+    } while (!subscriptionName);
+
+
+    return subs.find(sub => { return sub.displayName === subscriptionName });
+}
+
+async function acquireResourceGroup(subscription: SubscriptionModels.Subscription): Promise<ResourceGroup> {
+    //Acquire each subscription's data simultaneously
+    let resourceGroup;
+    let resourceGroupName;
+    const resourceGroupClient = new ResourceManagementClient(AzureCredentialsManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+    let resourceGroups = await AzureCredentialsManager.getInstance().getResourceGroups(subscription);
+
+    let resourceGroupNames: string[] = [];
+    resourceGroupNames.push('+ Create new resource group');
+    for (let i = 0; i < resourceGroups.length; i++) {
+        resourceGroupNames.push(resourceGroups[i].name);
+    }
+
+    do {
+        resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, { 'canPickMany': false, 'placeHolder': 'Choose a Resource Group to be used' });
+        if (resourceGroupName === undefined) throw 'user Exit';
+        if (resourceGroupName === '+ Create new resource group') {
```

Comment:
use const var

Reply:
won't these two functions also throw if the resourceGroupName/location isn't chosen?

Reply:
If it isn't chosen, the error message shows and the user is prompted once again